daemon/c_vol.c

```diff
@@ -1040,6 +1021,32 @@ c_vol_populate_dev_filter_cb(const char *dev_node, void *data)
 	return true;
 }
 
+static int
+c_vol_mount_dev(c_vol_t *vol)
+{
+	ASSERT(vol);
+
+	int ret = -1;
+	const char *mount_data = is_selinux_enabled() ? "rootcontext=u:object_r:device:s0" : NULL;
+	char *dev_mnt = mem_printf("%s/%s", vol->root, "dev");
+	int uid = container_get_uid(vol->container);
+	char *tmpfs_opts = (mount_data) ? mem_printf("uid=%d,gid=%d,%s", uid, uid, mount_data) :
+					  mem_printf("uid=%d,gid=%d", uid, uid);
+	if ((ret = mkdir(dev_mnt, 0755)) < 0 && errno != EEXIST) {
+		ERROR_ERRNO("Could not mkdir /dev");
+		goto error;
+	}
+	if ((ret = mount("tmpfs", dev_mnt, "tmpfs", MS_RELATIME | MS_NOSUID, tmpfs_opts)) < 0) {
+		ERROR_ERRNO("Could not mount /dev");
+		goto error;
+	}
```

Comment:
mem_free(dev_mnt);
mem_free(tmpfs_opts);

Reply:
this is two lines below 1045, 1046!

Reply:
right, i overlooked the falltrough